applications/vanilla/controllers/class.postcontroller.php

```diff
@@ -529,6 +530,12 @@ public function comment($DiscussionID = '') {
             $this->Form->addError(t('Failed to find discussion for commenting.'));
         }
 
+        // Vanilla Comments save as Text, no matter the format.
+        // Kludge to set format to Text if we're Wysiwyg to preserve newlines.
+        if ($isEmbeddedComments && ($this->Form->getFormValue('Format', c('Garden.InputFormatter')) === 'Wysiwyg')) {
```

Comment:
I was hoping to preserve the ability for a user to navigate to the forum and edit their post in the format specified by the editor. So, if we set every comment as Text and the editor format is BBCode, and if the user edits their comment on the forum and adds BBCode then BBCode won't render. WYSIWYG is the only format with the newline issue and there's a setting to reinterpret existing posts as WYSIWYG, so I think it's best to limit this to WYSIWYG.